### PR TITLE
added order applications by category in start menu

### PR DESCRIPTION
**Pull request by danielmeloalencar**
I understand that you plan to keep the core functionality in bond-wm and add the other features as separate packages. But I believe this contribution is still essential, I don't know if exactly in this way, but it should be something to consider.

This PR automatically organizes applications into categories for better viewing.

image

### CODE REVIEW

**Comment by wnayes** [comment]
Yeah this sort of change looks like it will be nice to have. The start menu is very much bare bones and could use improvements like these. I will take a closer look later.

**Comment by danielmeloalencar** [comment]
Do you want me to correct something so you can accept the PR? Or do you not accept contributions?

File: apps/wm/desktopEntries.ts [hunk 1]
```diff
@@ -152,11 +154,29 @@ async function parseDesktopEntries(desktopFolder: string): Promise<DesktopEntryM
       continue;
     }
 
+    const categoriesValue = desktopEntryGroupEntries["Categories"]?.value;
+    const categories = typeof categoriesValue === 'object' ? categoriesValue : ['Others'];
+
+    let assignedCategory = 'Others';
+    for (const category of categories) {
+      if (category !== 'Others' && existingCategories.has(category)) {
```
Comment (wnayes) [inline apps/wm/desktopEntries.ts:162]: It seems like `existingCategories` will be empty for the first entry, which could mean some entry gets assigned to Others incorrectly. The `existingCategories` set is potentially not needed if we instead say that `assignedCategory = categories[0]`.

File: apps/wm/desktopEntries.ts [hunk 2]
```diff
@@ -152,11 +154,29 @@ async function parseDesktopEntries(desktopFolder: string): Promise<DesktopEntryM
       continue;
     }
 
+    const categoriesValue = desktopEntryGroupEntries["Categories"]?.value;
+    const categories = typeof categoriesValue === 'object' ? categoriesValue : ['Others'];
```
Comment (wnayes) [inline apps/wm/desktopEntries.ts:158]: Would prefer `Array.isArray(categoriesValue) ? categoriesValue : ["Others"]`

**Comment by wnayes** [comment]
Can you run `pnpm prettier` before we merge this to update with matching formatting?

The following are not necessary to merge this, but some ideas in case you want to add them:

- There is a [standard set of categories](https://specifications.freedesktop.org/menu-spec/latest/category-registry.html) that we can expect. It would be nice if we had a map to map from identifier to display name, like `"AudioVideo" -> "Audio / Video"` and `"PackageManager" -> "Package Manager"`.
- Eventually, I might like this behavior to be controlled by a prop, e.g. `<StartMenuApplicationList groupBy="categories"  />`

**Review by wnayes — CHANGES_REQUESTED** [review]

**Comment by danielmeloalencar** [comment]
> Can you run `pnpm prettier` before we merge this to update with matching formatting?
> 
> The following are not necessary to merge this, but some ideas in case you want to add them:
> 
> * There is a [standard set of categories](https://specifications.freedesktop.org/menu-spec/latest/category-registry.html) that we can expect. It would be nice if we had a map to map from identifier to display name, like `"AudioVideo" -> "Audio / Video"` and `"PackageManager" -> "Package Manager"`.
> * Eventually, I might like this behavior to be controlled by a prop, e.g. `<StartMenuApplicationList groupBy="categories"  />`

Thanks for the feedback! I've already run pnpm prettier and made the recommended corrections. Everything should be in order now. I appreciate the additional suggestions and will keep them in mind for future improvements!

**Comment by danielmeloalencar** [comment]
Is everything ok?

**Comment by wnayes** [comment]
I am away from my computer this weekend, I will try to look by Sunday
night. Thanks for making the changes.

**Pull request merged**